### PR TITLE
feat: add support for app configuration with environment variables

### PR DESCRIPTION
**Pull request by ruimonteiro93**
Implements multiple environment variable files support and updates docs accordingly.

### CODE REVIEW

File: docusaurus/docs/welcome/instructions.md [hunk 1]
```diff
@@ -11,6 +11,24 @@ Keep in mind, part of adapting this boilerplate into a deliverable project is al
...
+
+#### Multiple environments
+
+Most of the times there will be a need to have multiple environment files. In order to have this, a bit of a setup is needed. It's a little different for each platform, but nothing too much to worry about.
+
+For `Android`, you need to go to `app/build.gradle` and change the `envConfigFiles`. Scroll down a bit in the file, and you shall find them. The boilerplate comes with three already (debug, staging and production) all pointing to `.env`, but you can change them to your liking. Try setting up a variable in `.env.sample`, and set it to `devdebug` in `app/build.gradle`, see if it works. It probably should, and you're ready to create as many environment files as you need.
+
+For `iOS`, go to Xcode and open your project.  Then, go to `Product` > `Scheme` > `Edit Scheme...`. In the Build section, go to `Pre-actions`. There, change `.env` to whatever file you just created, or are about to create. Maybe try it out with the already existing `.env.sample`, see how it goes. Repeat this process for the other two schemes if you wish (staging and production), and you're done.
```
Comment (acostalima) [inline docusaurus/docs/welcome/instructions.md:26]: There is no need to touch existing configurations. On both Android and iOS, react-native-config should look for `.env` by default regardless whether we're building for development, production, etc.

If the developer wants to have custom `.env.*` files for any particular reason, the following files are loaded up by the boilerplate's build process:
- .env.dev
- .env.prod
- .env.staging

Correct me if I'm wrong, but as far as my knowledge goes react-native-config works this way. The only thing I'm not 100% sure about concerning iOS is that if the file name written to `/tmp/envfile` does not exist, what will react-native-config do? Will it look for `.env` instead still? Can you confirm, please?

File: docusaurus/docs/welcome/instructions.md [hunk 2]
```diff
@@ -11,6 +11,24 @@ Keep in mind, part of adapting this boilerplate into a deliverable project is al
 
 ## 1. Tweak some project files
 
+### 1.1 Environment Configuration (.env)
```
Comment (acostalima) [inline docusaurus/docs/welcome/instructions.md:14]: I was thinking this through and I believe this whole section can be moved to "What's included > Environment variables". Using environment variables is entirely opcional, therefore it shouldn't be considered as instructions. Do you agree?

File: android/app/build.gradle [hunk 3]
```diff
@@ -81,9 +81,9 @@ project.ext.react = [
 ]
 
 project.ext.envConfigFiles = [
-    devDebug: ".env.dev",
-    prodRelease: ".env.prod",
-    stagingRelease: ".env.staging"
+    devdebug: ".env",
+    prodrelease: ".env",
+    stagingrelease: ".env"
```
[the remark this answers is not in the document]
Comment (ruimonteiro93) [reply android/app/build.gradle:86]: We need to remove this altogether if we want `.env` to be the default. If `extConfigFiles` are set here, react-native-config will always look for the files specified and not fallback to `.env`. We can leave clear instructions for this in the docs. Thoughts?

File: android/app/build.gradle [hunk 4]
```diff
@@ -231,6 +232,7 @@ android {
 }
 
 dependencies {
+    implementation project(':react-native-config')
```
[the remark this answers is not in the document]
Comment (ruimonteiro93) [reply android/app/build.gradle:235]: You're right. It's not needed here. 👍

File: android/app/build.gradle (hunk 3 above)
[the remark this answers is not in the document]
Comment (acostalima) [reply android/app/build.gradle:86]: I see. I'd say let's get rid of custom dotenvs entirely and stick to the default .env. The same should be done for iOS as well.

File: docusaurus/docs/welcome/instructions.md (hunk 1 above)
Comment (acostalima) [reply docusaurus/docs/welcome/instructions.md:26]: Following our discussion [above](https://github.com/moxystudio/react-native-with-moxy/pull/18#discussion_r403095139), I guess we still can point the developer to the use of `ENVFILE` environment variable to specify a custom `.env`

File: ios/reactNativeWithMoxy.xcodeproj/xcshareddata/xcschemes/reactNativeWithMoxy Staging.xcscheme [hunk 5]
```diff
@@ -10,7 +10,7 @@
             ActionType = "Xcode.IDEStandardExecutionActionsCore.ExecutionActionType.ShellScriptAction">
             <ActionContent
                title = "Run Script"
-               scriptText = "echo &quot;.env.staging&quot; &gt; /tmp/envfile&#10;">
+               scriptText = "echo &quot;.env&quot; &gt; /tmp/envfile&#10;">
```
[the remark this answers is not in the document]
Comment (acostalima) [reply ios/reactNativeWithMoxy.xcodeproj/xcshareddata/xcschemes/reactNativeWithMoxy Staging.xcscheme:13]: Following our discussion [above](https://github.com/moxystudio/react-native-with-moxy/pull/18#discussion_r403095139), please remove this scheme because it is no longer relevant.

File: ios/reactNativeWithMoxy.xcodeproj/xcshareddata/xcschemes/reactNativeWithMoxy Prod.xcscheme [hunk 6]
```diff
@@ -10,7 +10,7 @@
             ActionType = "Xcode.IDEStandardExecutionActionsCore.ExecutionActionType.ShellScriptAction">
             <ActionContent
                title = "Run Script"
-               scriptText = "echo &quot;.env.prod&quot; &gt; /tmp/envfile&#10;">
+               scriptText = "echo &quot;.env&quot; &gt; /tmp/envfile&#10;">
```
[the remark this answers is not in the document]
Comment (acostalima) [reply ios/reactNativeWithMoxy.xcodeproj/xcshareddata/xcschemes/reactNativeWithMoxy Prod.xcscheme:13]: Following our discussion [above](https://github.com/moxystudio/react-native-with-moxy/pull/18#discussion_r403095139), this script can be removed.

File: docusaurus/docs/welcome/instructions.md (hunk 1 above)
Comment (ruimonteiro93) [reply docusaurus/docs/welcome/instructions.md:26]: For multiple environment files, should we explain in the docs that the user will need to use the `ENVFILE` environment variable in the script to run the app, instead of using the pre-action scripts (iOS) or setting the `extConfigFiles` in build.gradel (Android) ?

Comment (acostalima) [reply docusaurus/docs/welcome/instructions.md:26]: I'd drop the "Multiple environments" section entirely to avoid confusion. I think it is enough to state the following:
- `.env` is loaded by default
- Briefly mention `ENVFILE` environment variable can be used to specify a custom dotenv file as instructed at https://github.com/luggit/react-native-config#different-environments
For everything else, forward the user to `react-native-config`'s docs.

What do you think?

Comment (ruimonteiro93) [reply docusaurus/docs/welcome/instructions.md:26]: That seems perfect to me actually. Should we leave this in instructions? Or do you want to still move it to What's Included?

File: ios/reactNativeWithMoxy.xcodeproj/xcshareddata/xcschemes/reactNativeWithMoxy.xcscheme [hunk 7]
```diff
@@ -10,7 +10,7 @@
             ActionType = "Xcode.IDEStandardExecutionActionsCore.ExecutionActionType.ShellScriptAction">
             <ActionContent
                title = "Run Script"
-               scriptText = "echo &quot;.env.dev&quot; &gt; /tmp/envfile&#10;">
+               scriptText = "echo &quot;.env&quot; &gt; /tmp/envfile&#10;">
```
[the remark this answers is not in the document]
Comment (acostalima) [reply ios/reactNativeWithMoxy.xcodeproj/xcshareddata/xcschemes/reactNativeWithMoxy.xcscheme:13]: Following our discussion [above](https://github.com/moxystudio/react-native-with-moxy/pull/18#discussion_r403095139), please remove this scheme because it is no longer relevant.

File: docusaurus/docs/welcome/instructions.md (hunk 1 above)
Comment (acostalima) [reply docusaurus/docs/welcome/instructions.md:26]: Yes, I think it makes more sense to move the whole document to "What's included > Environment variables". Unlike Next with MOXY, environment variables are not mandatory here. Therefore, I think we shouldn't give attention to this as far as project setup instructions go.

File: src/app/App.js [hunk 8]
```diff
@@ -4,15 +4,20 @@ import { ThemeProvider } from '../shared/components';
...
+    return (
+        <SafeAreaProvider>
+            <ThemeProvider>
+                <NavigationContainer ref={ rootNavigation.navigationRef }>
+                    <AppStack />
+                </NavigationContainer>
+            </ThemeProvider>
+        </SafeAreaProvider>
+    );
+};
```
Comment (acostalima) [inline src/app/App.js:21]: Please revert this change.

Comment (acostalima) [inline src/app/App.js:21]: Please revert this change.

File: package.json [hunk 9]
```diff
@@ -18,12 +18,11 @@
   "scripts": {
     "start:ios:dev": "react-native run-ios --configuration Debug",
     "start:ios:prod": "react-native run-ios --configuration Release",
-    "start:android:dev": "react-native run-android --variant devDebug",
-    "start:android:prod": "react-native run-android --variant prodRelease",
+    "start:android:dev": "react-native run-android --variant debug",
+    "start:android:prod": "react-native run-android --variant release",
     "build:android:js:dev": "react-native bundle --platform android --dev true --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle",
     "build:android:js:prod": "react-native bundle --platform android --dev false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle",
     "build:android:aab:prod": "cd android && ./gradlew bundleProdRelease || cd -",
```
Comment (acostalima) [inline package.json:25]: We should change this one too, sorry! 🙏 

```suggestion
    "build:android:aab:prod": "cd android && ./gradlew bundleRelease || cd -",
```

**Comment by acostalima** [comment]
This PR shouldn't be merged before #20.
